### ISSUE TITLE
Users are not distributed equally across workers.

### ISSUE THREAD

**Issue by guysv** [locustio/locust#2128]
### Describe the bug
Running locust in k8s across hundreds of workers, I peaked the workers UI tab and wondered why do some workers report to run several users, when most of the cluster is still not running a single user.

### Expected behavior
I expected the users to be equally distributed across my workers

### Actual behavior
I ran this test with 600 workers
<img width="1377" alt="image" src="https://user-images.githubusercontent.com/32539816/177377688-aea0859f-9de3-42c8-972d-4ece8d219d4c.png">

### Steps to reproduce
I guess it reproduces with a high number of workers?
 

### Environment

- OS: k8s
- Python version: locust latest official docker image
- Locust version: locust latest official docker image
- Locust command line that you ran: `args: ["-f", "/loadtest/locustfile.py", "--worker", "--master-host", "locust-master"]`
- Locust file contents (anonymized if necessary):
```
from locust import HttpUser, task
import subprocess

class LoadTestUser(HttpUser):
    @task
    def do_test(self):
        urlfile = subprocess.check_output(["bash", "-c", "find /loadtest/urlfiles -type f | shuf | head -n 1"]).rstrip()
        with open(urlfile) as f:
            urls = [l.rstrip() for l in f.readlines()]
        for u in urls:
            self.client.get(u, verify=False)
```

Basically the test fetches a random url list for a single flow in our system, and probes all the URLs.

**Comment by cyberw** [comment]
Probably this is a result of workers connecting after ramp up had started. Try setting --expect-workers to make locust wait until all workers have connected. Or maybe --enable-rebalancing but that might mess up long-running/stateful tasks.

**Comment by cyberw** [comment]
Ok! Perhaps your kubernetes setup scales up during the test? In that case —enable-rebalancing may be appropriate

**Comment by guysv** [comment]
No nodes were added during the test and as far as I can tell the workers have a good link. I tried that option on my master locust, but there are still idle workers when some of them reach even 3/4 users

**Comment by cyberw** [comment]
Yes, that master log indicates a problem. The worker logs are weird too, but I’m not sure what could cause that. Check network connection and cpu usage on workers and master. Possibly try increasing heartbeat timeout. Dont remember the exact variable though, and it is not exposed as a parameter if I remember correctly. But it should be in the docs.

**Comment by guysv** [comment]
OK. I modified `runners.py`
```diff
+HEARTBEAT_INTERVAL = 15
+HEARTBEAT_LIVENESS = 30
-HEARTBEAT_INTERVAL = 1
-HEARTBEAT_LIVENESS = 3
```

And the users really do get distributed more equally across the workers!
Unfortunately, I think those values also relate to the test data credibility because the RPS graph became alot more jittery after this patch.

So obviously the master is conjested, which made me rethink my test infrastructure. there is no reason for me to scale so many workers before attempting to consume my compute resources using a smaller number of workers.

**Comment by cyberw** [comment]
👍 one worker per core is optimal in most cases.

**Comment by cyberw** [comment]
> So obviously the master is conjested

well, I wouldnt be so sure, but it is very possible.
